src/components/swap/components/warning.tsx

```diff
@@ -49,17 +48,18 @@ export const WarningComp = (props: WarningProps) => (
 )
 
 export const WarningCompProtection = () => {
+  const ethereum = window.ethereum
+
   const onClick = async () => {
-    const ethereum = window.ethereum
-    if (!ethereum) return
     await addMEVProtectionChain(ethereum)
   }
+
+  if (!ethereum) return null
```

Comment:
Hide copy if user isn't using MetaMask

Reply:
Just FYI some wallets may inject via `ethereum` as well but I'd there is not much more we can check here.